addons/web_editor/static/src/js/widgets/widgets.js

```diff
@@ -550,6 +565,126 @@ var ImageWidget = MediaWidget.extend({
         });
         this.$el.submit();
     },
+    /**
+     * @private
+     */
+    _loadMoreImages: function () {
+        this.IMAGES_ROWS = this.IMAGES_ROWS + 2;
+        this.IMAGES_PER_PAGE = this.IMAGES_ROWS * this.IMAGES_PER_ROW;
+        this._renderImages();
+        this._displayLoadMoreBtnImages();
+    },
+    /**
+     * @private
+     */
+     _displayLoadMoreBtnImages: function () {
+        var records = this.records.length;
+        this.$('.o_load_more_unsplash').addClass('d-none');
+        if (this.IMAGES_PER_PAGE >= records) {
+            this.$('.o_load_more_images').addClass('d-none');
+            this.$('.o_load_done_msg').removeClass('d-none');
+        } else {
+            this.$('.o_load_more_images').removeClass('d-none');
+            this.$('.o_load_done_msg').addClass('d-none');
+        }
+    },
+    /**
+     * @private
+     */
+    _checkImageUrl: function () {
+        var format = ['.gif','.jpe','.jpg','.png'];
+        var url = this.$('#imageurl').val();
+        var acceptedURL = false;
+        var $warning = this.$('.warning-url-image');
+        var $success = this.$('.success-url-image');
+        var regex = /^(http:\/\/www\.|https:\/\/www\.|http:\/\/|https:\/\/)?[a-z0-9]+([\-\.]{1}[a-z0-9]+)*\.[a-z]{2,5}(:[0-9]{1,5})?(\/.*)?$/;
+
+        if (url.match(regex)) {
+            for (var i in format) {
+                var endsWith = url.endsWith(format[i]);
+                if (endsWith) {
+                    acceptedURL = true;
+                    $warning.toggleClass('d-none', true);
+                    $success.toggleClass('d-none', false);
+                    this._nameButton('image', 'images');
+                }
+            }
+        }
+        if (!acceptedURL) {
+            $warning.toggleClass('d-none', false);
+            $success.toggleClass('d-none', true);
+            this._nameButton('document', 'images');
+        }
+        if (url.length === 0) {
+            $warning.toggleClass('d-none', true);
+            $success.toggleClass('d-none', true);
+            this._nameButton('', 'images');
+        }
```

Comment:
Using `toggleClass` with plain boolean is not ok (there is `addClass` and `removeClass` for that). However, I suppose that the code should be factorized here, I suppose the method should finish with:
```javascript
$warning.toggleClass('d-none', <boolean_variable>);
$success.toggleClass('d-none', <boolean_variable>);
```
Do not write the DOM multiple times for nothing.